src/dropdown/dropdown.ts

```diff
@@ -513,6 +513,16 @@ export class NgbDropdown implements AfterContentInit, OnDestroy {
         renderer.removeClass(bodyContainer, 'dropdown');
         renderer.addClass(bodyContainer, dropdownClass);
       }
+
+      // We need to trim the value because custom properties can also include spaces
+      const isEnd = getComputedStyle(this._menu.nativeElement).getPropertyValue('--bs-position').trim() === 'end';
+      if (isEnd) {
+        renderer.setStyle(this._menu.nativeElement, 'left', 'auto');
```

Comment:
Instead of using setStyle, the appropriate css class should be added, right?

setting style attributes directly makes it harder to comply with certain security policies.

Reply:
@bbarry that's a good point. 

In bootstrap v4 the left and right properties were being set as part of the `dropdown-menu-right` class.
![image](https://user-images.githubusercontent.com/37119470/133849416-92177900-0b22-4e34-9e02-c7423c50da02.png)

However in V5 this class (renamed to `dropdown-menu-end`) has those styles [removed](https://getbootstrap.com/docs/5.0/components/dropdowns/#menu-alignment) 
![image](https://user-images.githubusercontent.com/37119470/133850159-484ed05c-9eb3-45c2-a9e8-97464b684131.png)

Instead of using the class to position the dropdown menu bootstrap v5 now relies on popper to do its positioning in this situation (see code [here](https://github.com/twbs/bootstrap/blob/23fd488c380c347a3d0121a49b6285bdd65418a2/js/src/dropdown.js#L285)).

This would be fine except that ng-bootstrap handles this a little differently when the dropdowns are placed within a navbar. The original issue #4100 was raised relating to the navbar dropdown in the docs. (see following image)
![image](https://user-images.githubusercontent.com/37119470/133850733-0a481942-56f1-4a63-a7c8-82b2803c4165.png)

In the case where the dropdown is placed inside a navbar, the display property gets set to a value of "static" which [ignores the placement configuration](https://github.com/ng-bootstrap/ng-bootstrap/blob/9598506dfcbb0112afe21a1d6157c4645fd47522/src/dropdown/dropdown.ts#L501).

Maybe we could add the `end-0` class to the element instead of the call to `setStyle`, but at this point I'm wondering if this would be better included as part of the popper2 migration work done on PR #4129. It might be  more inline with the way bootstrap handles this itself.

@fbasso dis you come across this issue with the static nav menu items when looking into integrating popper v2?